src/Oscoin/P2P/Disco.hs

```diff
@@ -102,18 +102,18 @@ withDisco tracer opt defaultGossipPort !advertise k = do
         DNS.makeResolvSeed
             . maybe identity
                     (\ns -> \rc -> rc { DNS.resolvInfo = uncurry DNS.RCHostPort ns })
-                    (optNameserver opt)
+                    (runIdentity (optNameserver opt))
```

Comment:
worthwhile moving `unI` inside `Oscoin.Prelude` so we can reuse it here as well?

Reply:
No. Should we move to `generics-sop`, the types won't match.